crates/red_knot_python_semantic/src/types.rs

```diff
@@ -120,6 +120,13 @@ fn symbol_by_id<'db>(db: &'db dyn Db, scope: ScopeId<'db>, symbol: ScopedSymbolI
 
 /// Shorthand for `symbol_by_id` that takes a symbol name instead of an ID.
 fn symbol<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Symbol<'db> {
+    match file_to_module(db, scope.file(db)) {
```

Comment:
Is it worth checking that `name` is equal to `"TYPE_CHECKING"` _before_ retrieving the module using `file_to_module()`? A simple string comparison should be much cheaper than the lookup in the salsa db, I think

Reply:
Yes — that also looks a bit cleaner now. Thanks.